youtube_dl/extractor/cda.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    decode_packed_codes,
+    ExtractorError,
+    parse_duration,
+)
+
+
+class CDAIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:(www|ebd)\.)?cda\.pl/(video|[0-9]+x[0-9]+)/(?P<id>[0-9a-z]+)'
+    _TEST = {
+        'url': 'http://www.cda.pl/video/5749950c',
+        'md5': '6f844bf51b15f31fae165365707ae970',
+        'info_dict': {
+            'id': '5749950c',
+            'ext': 'mp4',
+            'height': 720,
+            'title': 'Oto dlaczego przed zakrętem należy zwolnić.',
+            'duration': 39,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage("http://ebd.cda.pl/0x0/" + video_id, video_id)
+
+        if 'Ten film jest dostępny dla użytkowników premium' in webpage:
+            raise ExtractorError('This video is only available for premium users.', expected=True)
+
+        title = self._html_search_regex(r'<title>(.+?)<\/title>', webpage, 'title', fatal=False)
+        duration = [None]
+
+        def _get_format(page):
+            unpacked = decode_packed_codes(page)
+            if duration[0] is None:
+                duration[0] = self._search_regex(r"duration:\\'(.+?)\\'", unpacked, 'duration', default=None)
+
+            m = re.search(r'<a data-quality="(?P<format_id>[^"]+)" href="[^"]+" class="quality-btn quality-btn-active">(?P<height>[0-9]+)p<\/a>',
```

Comment:
This is expected. `_get_format` is called from within the loop where there are already matched buttons for each format so we know that they are available. We just "select" format we know it's available.

I can add safe check here if you prefer, but the point is to call `_get_format` on well-formed pages and if this is not the case it is likely that the issue is somewhere else.